fetcher/fetcher.go

```diff
@@ -14,19 +14,18 @@ import (
 
 // Fetcher for task statuses by contacting task management web apps' rest api
 type Fetcher struct {
-	origin         string
-	tracker        config.IssueTracker
+	issuetracker.IssueTracker
 	authMiddleware authmiddleware.Func
 }
 
 // NewFetcher instance
-func NewFetcher(origin string, tracker config.IssueTracker, authMw authmiddleware.Func) *Fetcher {
-	return &Fetcher{origin, tracker, authMw}
+func NewFetcher(issueTracker issuetracker.IssueTracker, authMw authmiddleware.Func) *Fetcher {
+	return &Fetcher{issueTracker, authMw}
 }
 
 // Fetch a task's status based on task ID
 func (f *Fetcher) Fetch(taskID string) (taskstatus.TaskStatus, error) {
-	req, err := http.NewRequest("GET", f.origin+issuetracker.TaskURLSuffixFor(taskID, f.tracker), nil)
+	req, err := http.NewRequest("GET", f.IssueTracker.IssueAPIOrigin()+f.IssueTracker.TaskURLFrom(taskID), nil)
```

Comment:
I think it would be better if `IssueTracker` provided an abstract method and then here we didn't need to concatenate these two.

Reply:
You're right, I agree this is a good idea.

Does `IssueURLFor(taskID string)` sound like a good name?

Additionally, I also suggest doing this in a follow-up PR to avoid too many changes from refactoring the original approach.

Reply:
Sure! the name is good enough.

So I'll resolve this one since it'll be addressed in another PR.